### PR TITLE
extend ListView.DataSource

### LINKED ISSUE #3: Incompatible with RN 0.31.0
It seems like the new version of react-native included some changes that enforce this requirement:
https://github.com/facebook/react-native/blob/bd60d828c5fc9cb066e5f647c87ecd6f70cb63a5/Libraries/CustomComponents/ListView/ListView.js#L131
which has been present for long time in the `ListView`

I am not sure if there is an easy fix to make this module extend `ListViewDataSource` or this is simply not possible anymore.

screenshot_2016-08-12-14-34-21

### PR DESCRIPTION
**Pull request by fabriziomoscon**
this changes takes the improved functionalities of `1.1.1` and the correct extension of `1.0.0`

- fixes #3

### CODE REVIEW

**Comment by DaleJefferson** [comment]
That's for the PRs, I had the same idea originally of extending DataSource its much simpler like that, sadly it does not work, super.cloneWithRows calls new ListView.DataSource not new this. I tried to land a PR in React Native to fix this but it was rejected. 

When you call cloneWithRows a second time to update the list it will not work. 

With your code the next time you call

**Comment by DaleJefferson** [comment]
This was my PR https://github.com/facebook/react-native/pull/5602/files

**Comment by DaleJefferson** [comment]
In the example I pop an country from the list to exercise this functionality, you should be able to see it fail.

**Comment by fabriziomoscon** [comment]
I think I am missing something. By the sound of that comment from the Facebook guy it seems like this module is completely unnecessary. Is it the case?
Initially I liked the idea that code like this one: `(r1, r2) => r1 !== r2` would be included in a common file but if that is not possible...

Maybe the solution is to have a MyListView component that renders ListView using for datastore the functions you override in this module?

**Comment by fabriziomoscon** [comment]
Very much related: https://github.com/facebook/react-native/issues/3544

**Comment by fabriziomoscon** [comment]
I have created a wrapper for <ListView> as suggested by the guys at Facebook.
The `countries.pop()` in the example works. In this way it is possible to keep custom functions to compare rows and sections inside the module avoiding leaking into the application code. And the problem with subclassing is avoided.

Let me know if you see any problem with this. And possibly how to make the Example do more operation rather than only pop to prove that is actually works.
I have implemented in my app for 6 list views and it seems working well.

Note: something should be done to make the component name match the functionality

Let me know what do you think.

**Comment by DaleJefferson** [comment]
Thanks for your input @fabriziomoscon providing a ListView instead of a ListView.DataSource was another one of the solutions I tried and it works very nicely. I have a version of this in a private repo, this library is called react-native-immutable-listView-dataSource not react-native-immutable-listView so I don't think this is correct place to host this solution. Fell free to publish your version on npm but I don't think it makes sense to merge it into this project without renaming.

I will fix the 0.31 issues and add section support to this library as well as improve the example and unit test coverage.

File: Example/CountiesByPopulation.js [hunk 1]
```diff
@@ -5,14 +5,13 @@ import React, {Component} from 'react'
...
     StyleSheet,
-    ListView,
     Text,
     View
 } from 'react-native'
 
 import immutable from 'immutable'
 
-import ImmutableDataSource from 'react-native-immutable-listview-datasource'
+import MyListView from 'react-native-immutable-listview-datasource'
```
Comment (DaleJefferson) [inline Example/CountiesByPopulation.js]: This should be

```
import MyListView from 'react-native-immutable-listview'
```

But that would be a different library.

**Comment by fabriziomoscon** [comment]
ok, I thought it was going into another direction.
I will do as you suggested. And keep an eye open for the changes will be making in here.

Thanks

**Pull request closed without merging**